### PR TITLE
Fix ranger multi-advertise

### LINKED ISSUE #2480: DISTANCE_SENSOR multi-subscribe still broken
I've been trying to get the PX4Flow sonar + LL40LS working together. 

```ls /dev/``` gives me :
```
 range_finder0
 range_finder1
```
I can see that the PX4flow sonar gets instance ID 0 ( Can see data in QGC )
I assume that the ll40ls is getting uorb instance ID 1, but using orb_subscribe_multi() with the instance set to 1 doesn't give me any data.

Any insights?

### PR DESCRIPTION
**Pull request by mhkabir**
This fixes #2480. 

I tested this with a LL40LS + PX4Flow sonar module.

### CODE REVIEW

File: src/drivers/mb12xx/mb12xx.cpp [hunk 1]
```diff
@@ -270,16 +270,14 @@ MB12XX::init()
 
 	_class_instance = register_class_devname(RANGE_FINDER_BASE_DEVICE_PATH);
 
-	if (_class_instance == CLASS_DEVICE_PRIMARY) {
-		/* get a publish handle on the range finder topic */
-		struct distance_sensor_s ds_report = {};
+	/* get a publish handle on the range finder topic */
+	struct distance_sensor_s ds_report;
```
Comment (LorenzMeier) [inline src/drivers/mb12xx/mb12xx.cpp]: This is missing the zero-initialiser braces, which will result in random memory content being published.

File: src/drivers/px4flow/px4flow.cpp [hunk 2]
```diff
@@ -241,16 +241,14 @@ PX4FLOW::init()
 
 	_class_instance = register_class_devname(RANGE_FINDER_BASE_DEVICE_PATH);
 
-	if (_class_instance == CLASS_DEVICE_PRIMARY) {
-		/* get a publish handle on the range finder topic */
-		struct distance_sensor_s ds_report = {};
+	/* get a publish handle on the range finder topic */
+	struct distance_sensor_s ds_report;
```
Comment (LorenzMeier) [inline src/drivers/px4flow/px4flow.cpp]: Missing zero initialiser braces.

File: src/drivers/sf0x/sf0x.cpp [hunk 3]
```diff
@@ -296,16 +296,14 @@ SF0X::init()
 
 		_class_instance = register_class_devname(RANGE_FINDER_BASE_DEVICE_PATH);
 
-		if (_class_instance == CLASS_DEVICE_PRIMARY) {
-			/* get a publish handle on the range finder topic */
-			struct distance_sensor_s ds_report = {};
+		/* get a publish handle on the range finder topic */
+		struct distance_sensor_s ds_report;
```
Comment (LorenzMeier) [inline src/drivers/sf0x/sf0x.cpp]: Missing zero initialiser braces.

File: src/drivers/srf02/srf02.cpp [hunk 4]
```diff
@@ -269,16 +269,14 @@ SRF02::init()
 
 	_class_instance = register_class_devname(RANGE_FINDER_BASE_DEVICE_PATH);
 
-	if (_class_instance == CLASS_DEVICE_PRIMARY) {
-		/* get a publish handle on the range finder topic */
-		struct distance_sensor_s ds_report = {};
+	/* get a publish handle on the range finder topic */
+	struct distance_sensor_s ds_report;
```
Comment (LorenzMeier) [inline src/drivers/srf02/srf02.cpp]: Same.

**Comment by LorenzMeier** [comment]
In general it would be preferable to publish a first value, but at the very least the struct should be set to zero before advertising.

**Comment by mhkabir** [comment]
@LorenzMeier Ah. I had to remove it in order to make it compile. The goto statements were creating issues when the struct was inititalized: 
```
/home/kabir/PX4Firmware/src/drivers/px4flow/px4flow.cpp: In member function 'virtual int PX4FLOW::init()':
/home/kabir/PX4Firmware/src/drivers/px4flow/px4flow.cpp:257:1: error: jump to label 'out' [-fpermissive]
 out:
 ^
```

**Comment by LorenzMeier** [comment]
Just initialise above the goto:
https://github.com/mhkabir/Firmware/blob/ranger_multi/src/drivers/mb12xx/mb12xx.cpp#L255

In general be vigilant about solving compile errors correctly, be sure that you understand the reason for the warning and the correct fix.

**Comment by LorenzMeier** [comment]
@mhkabir It would be good if you could fix trivial things like these in a timely fashion. Sending pull requests is great, but the idea behind the model is to free bandwidth on the maintainer side so the project can achieve more. If the amount of time to review and keep track of a PR is bigger than doing the associated changes it consumes critical bandwidth of the overall project.

Thanks!

**Comment by mhkabir** [comment]
@LorenzMeier Done. I just removed the gotos completely. Hope this is OK with you.

**Comment by LorenzMeier** [comment]
I'll do it myself. You just killed the proper error handling and resource cleanup.

**Comment by LorenzMeier** [comment]
Sorry, was too quick.

**Comment by LorenzMeier** [comment]
It helps if you indicate why you change, in this case because the gotos had no real function. In general a PR should state the intention so I can check if the changes look and match the intentions. In particular if it ends up doing structural changes like this one. Looks good!

**Comment by mhkabir** [comment]
Ok :)
I was slowly roasting myself here trying to figure out where I had messed up.

**Comment by LorenzMeier** [comment]
Yes, my apologies. I just had looked at the diff which sent me in the completely wrong direction.

**Comment by LorenzMeier** [comment]
Rebased and merged.

**Pull request closed without merging**